src/content_scripts/adblocker.js

```diff
@@ -9,15 +9,124 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0
  */
 
-import { injectCosmetics } from '@ghostery/adblocker-webextension-cosmetics';
+function debounce(fn, { waitFor, maxWait }) {
+  let delayedTimer;
+  let maxWaitTimer;
 
-function getCosmeticsFilters(args) {
-  chrome.runtime.sendMessage({
-    action: 'getCosmeticsFilters',
-    ...args,
-  });
+  const clear = () => {
+    clearTimeout(delayedTimer);
+    clearTimeout(maxWaitTimer);
+    delayedTimer = undefined;
+    maxWaitTimer = undefined;
+  };
+
+  const run = () => {
+    clear();
+    fn();
+  };
 
-  return Promise.resolve({});
+  return () => {
+    if (maxWaitTimer === undefined) {
```

Comment:
Why change the original implementation (omitting the `maxWait > 0` check)?

https://github.com/ghostery/adblocker/blob/a7d149e36bb7dc748b3081eeb8fc14858210164e/packages/adblocker-content/src/index.ts#L62

Reply:
IHMO, this change makes the function worse. Before, maxWait was an optional parameter, now it is is not (it will do setTimeout(..., undefined), which fires directly.

Reply:
The logic is exactly the same, the `maxWait` is always grether than 0...

```js
      if (maxWait > 0 && maxWaitTimer === undefined) {
```

Reply:
Note that "debounce" implementation normally has no mandatory maxWait, so we are deviating from the standard interface by making it mandatory. The existing implementation would also support calls like:

```
debounce(fn, { waitFor: 25  });
```

I'm aware that in our current heuristic, we do pass a `maxWait`. (When we experimented, we originally did not have it. I was in favor of adding such an enforcement, so this is where it came from. However, even for me, it is not clear if it is better or worse to enforce every one second if there were pending changes recently.)

Well, I'm still not convinced that optimizing here makes sense. The old implementation was fine in my opinion. Code also get reused and requirements change. But if you feel it is making it more readable by making it a required parameter, we can change the interface.